src/resource-group.cxx

```diff
@@ -13,86 +13,86 @@
 
 NAMESPACE__THITTAM__START
 
-bool
-ResourceGroup::validate_resources_id (const std::string & resource_id)
+Resource &
+ResourceGroup::_get_resource (const int index)
 {
-  for (auto & it : m_resources)
-  {
-    if (it.id() == resource_id)
-    {
-      return false;
-    }
-  }
-  return true;
+  return *std::next(m_resources.begin (), index);
 }
 
-std::string
-ResourceGroup::generate_unique_resource_id (void)
+const Resource &
+ResourceGroup::get_resource (const int index) const
 {
-  // In case number of resources is more than MOD_MAX
-  // generating unique id would not be possible
-  if (m_resources.size() < MOD_MAX)
+  return *std::next(m_resources.begin (), index);
+}
+
+void
+ResourceGroup::renumber_ids (void)
+{
+  for (auto & it : m_resources)
   {
-    unsigned int seeder = 0;
-    std::string id = std::to_string(util::generate_random_id());
-    for (; !ResourceGroup::validate_resources_id(id) ; )
-    {
-      id = std::to_string(util::generate_random_id() + seeder);
-      seeder++;
-    }
-    std::cout << "Executed " << seeder << " times\n\n";
-    return id;
+    id_counter++;
+    it.set_id(std::to_string(id_counter));
   }
-  return FAILED_TO_GENERATE_RANDOM_ID;
 }
 
-Resource &
-ResourceGroup::_get_resource (const int index)
+void
+ResourceGroup::create_resource_id (void)
 {
-  return *std::next(m_resources.begin (), index);
+  if (id_counter - m_resources.size() > SIZE_COUNTER_DIFFERENCE)
```

Comment:
Make this a `signed` comparison.

Reply:
there exist no scenario when id_counter < m_resource.size(). so no need of abs(id_counter - m_resources.size())